ens/utils.py

```diff
@@ -267,32 +270,38 @@ def get_abi_output_types(abi: 'ABIFunction') -> List[str]:
 # -- async -- #
 
 
-async def async_init_web3(
-    provider: "BaseProvider" = cast("BaseProvider", default),
-    middlewares: Optional[Sequence[Tuple["Middleware", str]]] = None
+def init_async_web3(
```

Comment:
Note: I did not start the method name with `async` on purpose since this method _is not_ `async`... but rather it initializes an async-compatible web3 instance. I'm a bit more used to anything that starts with `async` having to be awaited, which is not the case here.